qiskit/__init__.py

```diff
@@ -78,7 +78,7 @@
 __path__ = pkgutil.extend_path(__path__, __name__)
 
 # Please note these are global instances, not modules.
-from qiskit.providers.basicaer import BasicAer
+from qiskit.providers.basic_provider import BasicProvider
```

Comment:
I think that's fair, I went ahead and looked for the exact 1-1 correspondence but I agree it doesn't make much sense in the new context (this will be the answer for most of your comments).

Reply:
At this point, if we get rid of the alias, we might as well change the use of basic aer in the tests (and docs) from always doing

```
from qiskit import BasicAer
sim = BasicAer.get_backend("bla_simulator")
```
to 
```
from qiskit.providers.basic_provider import BasicSimulator
sim = BasicSimulator()
```

(moving away from the use of the provider). I feel like it's much clearer and easier to communicate.

Reply:
Yeah, I think that makes sense

Reply:
Done in [2ac10d0](https://github.com/Qiskit/qiskit/pull/11583/commits/2ac10d04cb222483787ad07de433ea62c66f7912)